numojo/routines/creation.mojo

```diff
@@ -995,6 +995,26 @@ fn fromstring[
     return result^
 
 
+# Technically we should allow for runtime type inference here, but NDArray doesn't support it yet.
+fn astype[dtype: DType](a: NDArray) raises -> NDArray[dtype]:
```

Comment:
There is an `NDArray.astype` already. So I think it is better to do `return a.astype[dtype]` here, so that we just need to maintain one place.

Reply:
I have done this in a commit.

Reply:
ah, but numpy has an explicit function `asarray()` as part of it's manipulation routines. So shouldn't we keep this?